src/topology/mod.rs

```diff
@@ -209,14 +204,29 @@ impl RunningTopology {
 
             if success.is_ok() {
                 info!("All healthchecks passed");
+
+                if exit_after == Some(Stage::HealthChecksPassed) {
+                    return true;
+                }
             } else {
                 error!("Sinks unhealthy");
                 return false;
             }
         } else {
+            debug_assert!(exit_after != Some(Stage::HealthChecksPassed));
```

Comment:
is there a way we can avoid this assert statically? We can probably include this conditionaly check in the if statement on line 202.

Reply:
This condition is always true, I put it here just for documentation and self-check purpose.
I can remove it if it hinders readability.

Reply:
Removed after current refactoring (stages gone, replaced with `--dry-run` + `--require-healthy`).